### PR TITLE
Updates

### PR DESCRIPTION
**Pull request by Gold856**

### CODE REVIEW

File: grammarGenerator.js
Comment (23watejona) [inline grammarGenerator.js:1]: Is there a better way to do this? (There is...)

File: index.js
Comment (23watejona) [inline index.js:1]: These are intentionally separated - you can reuse the command list for multiple files with different methods in each

File: test/test.js [hunk 1]
```diff
@@ -4,8 +4,7 @@ import assert from 'node:assert/strict';
```
Comment (23watejona) [inline test/test.js:3]: See comment on index

File: test/testScript.txt [hunk 2]
```diff
@@ -92,7 +92,6 @@ getOuttakePieceCommand
 		gripper WheelGripperPosition.OUTTAKE
 		wait 0.5
 		gripper WheelGripperPosition.STOP
-
```
Comment (23watejona) [inline test/testScript.txt]: not sure exactly why this file is changed

File: .gitattributes [hunk 3]
```diff
@@ -1,2 +1,2 @@
 # Auto detect text files and perform LF normalization
-* text=auto
```
Comment (23watejona) [inline .gitattributes:2]: why?...

**Review by 23watejona — CHANGES_REQUESTED** [review]

File: .gitattributes (hunk 3 above)
Comment (Gold856) [reply .gitattributes:2]: `* text=auto` doesn't force LF line endings. I have `core.autocrlf` enabled in my Git settings, so it auto converted it to CRLF's. To force LF's, you need the extra `eol=lf` bit

File: test/testScript.txt (hunk 2 above)
Comment (Gold856) [reply test/testScript.txt]: The tests failed without it. Sidenote, your error report doesn't give an accurate line number of where the error is.

File: grammarGenerator.js
Comment (Gold856) [reply grammarGenerator.js:1]: What's the problem here? It's not that bad and it works. The tests pass, and arrays are easier to type in than key-value pairs.

Comment (23watejona) [reply grammarGenerator.js:1]: I meant to do the parameter naming. Hint: use the existing array.map

Comment (Gold856) [reply grammarGenerator.js:1]: There is a much better way to do that

File: index.js [hunk 4]
```diff
@@ -27,7 +27,7 @@ export function generateGrammar(commandList, functionList) {
 	functionList = JSON.parse(functionList)
 	
 	//Generate the grammar
-	return grammarGenerator.generateGrammar(commandList, functionList)
+	return grammarGenerator.generateGrammar(commandList.commands, functionList)
```
Comment (23watejona) [inline index.js]: Why is this necessary?

File: test/testScript.txt (hunk 2 above)
Comment (23watejona) [reply test/testScript.txt]: I don't like this answer - can you figure out why this is?

**Review by 23watejona — CHANGES_REQUESTED** [review]

File: index.js (hunk 4 above)
Comment (Gold856) [reply index.js]: Commands are wrapped in a root `commands` key.

Comment (23watejona) [reply index.js]: But why? It's implied in the file

**Review by 23watejona — CHANGES_REQUESTED** [review]

File: test/testScript.txt (hunk 2 above)
Comment (Gold856) [reply test/testScript.txt]: Well I can tell you the line number inaccuracy is due to new lines in Java blocks not being counted

Comment (23watejona) [reply test/testScript.txt]: I know the line errors are wrong... I meant why you needed to change the spacing there

File: test/testcommandlist.json [hunk 5]
```diff
@@ -1,166 +1,168 @@
 {
```
Comment (23watejona) [inline test/testcommandlist.json:1]: Remove unnecessary wrapper plz

**Review by 23watejona — CHANGES_REQUESTED** [review]

**Review by 23watejona — APPROVED** [review]

**Pull request merged**